### ISSUE TITLE
mount documentation does not match code

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#30206]
From @ozkatz on 2014-12-08T16:38:18Z

**Issue Type:**

Bug Report
Documentation Report

**Component Name:**
mount module

**Ansible Version:**

$ ansible --version
ansible 1.6.1 _[@ozkatz: also true in 1.8.2]_

**Environment:**

N/A

**Summary:**

I'm reopening issue [#7791 from the main ansible repo](https://github.com/ansible/ansible/issues/7791) by @mpdehaan's instructions.
I'll paste the original issue as is - I hope that's ok with @shaunbrady (who opened the original issue).

---

There seems to be a mismatch with regard to the documentation/behavior of the mount modules.

Looking from line https://github.com/ansible/ansible-modules-core/blob/devel/system/mount.py#L277
we see that "absent" is suppose to "remove from fstab and unmounted", and the code appears to reflect this. As well "unmount" is suppose to "not change fstab state, but unmount" and the code appears to reflect this.

On the documentation page http://docs.ansible.com/mount_module.html we read:

"If mounted or unmounted, the device will be actively mounted or unmounted as well as just configured in fstab. absent and present only deal with fstab. mounted will also automatically create the mount point directory if it doesn't exist. If absent changes anything, it will remove the mount point directory."

The code and documentation for "unmount" and "absent" are in contention.

**Steps To Reproduce:**

Review code/documentation at provided links.

**Expected Results:**

I expect that the documentation would be the correct way for the software to behave. It seems to be the most extensible. Either way, consistency is more important.

**Actual Results:**

The code and documentation for "unmount" and "absent" are in contention.

I would be happy to provide a pull request for documentation or for code, but I wanted to see which (or both) were to be changed.

Thank you.

SB

Copied from original issue: ansible/ansible-modules-core#464

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (100/100 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 7188 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:54889 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drqMU-002VHu-Cq
	for greg@biomassiv.es; Tue, 12 Sep 2017 13:56:42 -0500
Date: Tue, 12 Sep 2017 11:56:02 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505242562;
	bh=Lt3C2dmwKjjgDpZaH3mzwMSXB1hmf28aHyNa7sW1izo=;
	h=From:Reply-To:To:Cc:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=VHuH88jYzFJJuZGDPHEaJW56yic9h9G4aVi7z+OVL27gkTxQnszjcjuF0ZCbdDeUb
	 jWme9DRJbB7zGcXLO5QkNeISgZYFg0bgiToC5wtKlL7rwdHSDh8EJPPkwpKEvJvqyR
	 0diJT1PMOqLLGOx5/Tc0f67MbM3m6Eb+FwNIinEw=

**Comment by ansibot** [comment]
From @ansibot on 2014-12-08T16:38:18Z

# Can You Help Us Out?

Thanks for filing a ticket! I am the friendly GitHub Ansibot. 

It looks like you might not have filled out the issue description based on our standard issue template.  You might not have known about that, and that's ok too, we'll tell you how to do it.

We have a standard template because Ansible is a really busy project and it helps to have some standard information in each ticket, and GitHub doesn't yet provide a standard facility to do this like some other bug trackers.  We hope you understand as this is really valuable to us!. 

Solving this is simple: please copy the contents of this [template](https://raw.githubusercontent.com/ansible/ansible/devel/ISSUE_TEMPLATE.md) and **paste it into the description** of your ticket.  That's it!

# If You Had A Question To Ask Instead

If you happened to have a "how do I do this in Ansible" type of question, that's probably more of a user-list question than a bug report, and you should probably ask this question on the project mailing list instead.  

However, if you think you have a bug, the report is the way to go!  We definitely want all the bugs filed :)  Just trying to help!

# About Priority Tags

Since you're here, we'll also share some useful information at this time.

In general tickets will be assigned a priority between P1 (highest) and P5, and then worked in priority order. We may also have some follow up questions along the way, so keeping up with follow up comments via GitHub notifications is a good idea.

Due to large interest in Ansible, humans may not comment on your ticket immediately.  

# Mailing Lists

If you have concerns or questions, you're welcome to stop by the ansible-project or ansible-development mailing lists, as appropriate.  Here are the links:
- https://groups.google.com/forum/#!forum/ansible-project - for discussion of bugs and how-to type questions
- https://groups.google.com/forum/#!forum/ansible-devel - for discussion on how to implement a code change, or feature brainstorming among developers

Thanks again for the interest in Ansible!

**Comment by ansibot** [comment]
From @ansibot on 2014-12-08T16:38:18Z

@grossws, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (101/100 (101%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 9115 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:49042 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drqNo-002Vyu-10
	for greg@biomassiv.es; Tue, 12 Sep 2017 13:58:04 -0500
Date: Tue, 12 Sep 2017 11:57:24 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505242644;
	bh=vioxGDjraHRZeTjMgd1ZL2RjahWCSkfKkhcLP3ZJ38Y=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=Jc+8g/9YCxbpfkgvMzAiPBVN9W4rSjqhb1oBZ4bnpqHByDWDbPsWDQnjAO6Av4AbV
	 9JHiT1ToOm7YDcr8L6o6GD00nDPTE0ExOC+iRz96Wqaxpa9w7SKXYAyco1KWkhkviK
	 /TIuswsbjK2w3br01sGfSxD2qTaJZEqJdqWQzD1E=

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (102/100 (102%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:39125 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drqO0-002W56-ES
	for greg@biomassiv.es; Tue, 12 Sep 2017 13:58:16 -0500
Date: Tue, 12 Sep 2017 11:57:36 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505242656;
	bh=DvoP+GYRrB+NEFvV0RViVkJC7GlF16ail0nPSpIy+gI=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=NfkakZ9DMwRDI3fXh8gCaXsL0lcL/XyBRkCblJhtqDBUUG7qp7SZENhTtFHNG9LCN
	 CtiVRMbR9z/Z75DmWR84dxoUxftQI35jLaXgmFcGykHtCEbShbIpwOguSEhhN+T1jW
	 Qt3hw/OocPYgGeuHJpj7q8l+UDA5mWO7tsgMLP7g=

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jtyr
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtyr** [comment]
I believe this issue has been fixed already as the `state` description in the module documentation is different now.

`close_me`.

**Comment by shaunbrady** [comment]
Can this be reopened?  `unmount` still appears to be in contention?  I can submit a PR if some one agrees.

**Comment by shaunbrady** [comment]
I am incorrect.  I was looking at older documentation by accident.  My apologies.
